src/main/java/pl/wojciechkostecki/revel/service/MenuService.java

```diff
@@ -0,0 +1,60 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.exception.BadRequestException;
+import pl.wojciechkostecki.revel.mapper.MenuMapper;
+import pl.wojciechkostecki.revel.model.Local;
+import pl.wojciechkostecki.revel.model.Menu;
+import pl.wojciechkostecki.revel.model.dto.MenuDTO;
+import pl.wojciechkostecki.revel.repository.MenuRepository;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@Transactional
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final LocalService localService;
+    private final MenuMapper menuMapper;
+
+    public MenuService(MenuRepository menuRepository, LocalService localService, MenuMapper menuMapper) {
+        this.menuRepository = menuRepository;
+        this.localService = localService;
+        this.menuMapper = menuMapper;
+    }
+
+    public Menu save(MenuDTO menuDTO) {
+        Menu menu = menuMapper.toEntity(menuDTO);
+        Local local = localService.findById(menuDTO.getLocalId());
+        if (Objects.nonNull(local.getMenu())) {
+            throw new BadRequestException(local.getMenu().getId(), menuDTO.getLocalId());
+        }
+        menu.setLocal(local);
+        local.setMenu(menu);
```

Comment:
I dont think this line does anything actually since you are not saving the local and I dont think its necessary. I think its safe to remove it.

Reply:
With this line I set the menu for the local. I removed it for a trial - I created the local, then a menu where I entered the id of the local to which it is to be assigned and without this line it was not assigned to the local.  Must stay :)

Reply:
That is not true for sure. You are not even saving updated local so you invoking `local.setMenu` doesnt change anything in the DB